Restore v:shell_error in autocmd ShellCmdPost

nkouevda opened the pull request:
Fixes #869.

airblade commented:
I think it might be better to move this save-and-restore logic to where `v:shell_error` gets set, i.e. somewhere around here: https://github.com/airblade/vim-gitgutter/blob/5f525476f3ade40e60aa295a8a2a01c0935b0d61/autoload/gitgutter/diff.vim#L93.

nkouevda commented:
> I think it might be better to move this save-and-restore logic to where `v:shell_error` gets set, i.e. somewhere around here:

Ah, I was hoping to also handle the case where 1) `g:gitgutter_async && gitgutter#async#available()` is false, 2) the file we're editing _does_ exist in `g:gitgutter_diff_base`, and 3) we run something like `:!exit 1`. For that we'd need to do the same `v:shell_error`-restoration around [the actual `git diff`](https://github.com/airblade/vim-gitgutter/blob/5f525476f3ade40e60aa295a8a2a01c0935b0d61/autoload/gitgutter/diff.vim#L179).

So instead of duplicating this logic, I've moved it into `gitgutter#utility#system` and updated all callers; let me know if that's acceptable, or if you'd prefer doing this only in the 2 places we actually need it in `diff.vim`.

airblade commented:
Thanks, that's perfect.
